docs/guide/advanced/accessibility.md

```diff
@@ -1,4 +1,204 @@
+screen readers used by people with visual impairments.
+
+However: descriptions should only be provided to standalone icons that aren't purely decorative, as
+providing accessible names to non-functional elements only increases clutter when using screen
+readers.
+
+### On standalone icons
+
+Icons are usually very unlikely to stand on their own with no semantically meaningful wrapper
+element. In most cases they will be part of a badge, button (including icon buttons), navigation
+item or other interactive UI element.
+
+::: warning
+In case some of your icons stand alone, and they serve a non-decorative function, make sure to
+provide the appropriate accessible label for them.
+:::
+
+![In short: provide accessible label for semantic icons, but not for decorative icons.](../../images/a11y/alttext-standalone.svg?raw=true)
+
+In general try to avoid using functional icons with no interactivity, we recommend that:
+
+1) you either add a visible description next to them, or
+2) place them in badges, labels or on buttons, and at least add a tooltip to them.
+
+In any such case, it is preferred that the accessible name be provided for these interactive
+elements (badges, buttons, nav items etc.) only, _not_ the icons themselves.
+
+### On buttons
+
+Do not provide an accessible label to icons when used on a button, as this label will be read out by
+screen readers, leading to nonsensical text.
+
+![](../../images/a11y/alttext-buttons.svg?raw=true)
+
+::: details Code examples
+
+```tsx
+// Don't do this
+<button>
+  <Plus aria-label="Plus icon"/>
+  Add document
+</button>
+
+// Do this, just leave it
+<button>
+  <Plus/>
+  Add document
+</button>
+```
+
+:::
+
+### On icon buttons
+
+Icon buttons are buttons that do not contain any visible text apart from the icon itself (think of
+the close button of a dialog for example).
+
+As previously stated, you should provide your accessible label on the icon button itself, not the
+contained icon.
+
+![](../../images/a11y/alttext-iconbuttons.svg?raw=true)
+
+::: details Code examples
+
+```tsx
+// Don't do this
+<button class="btn-icon">
+  <Home/>
+</button>
+
+// Don't do this either
+<button class="btn-icon">
+  <Home aria-label="Home icon"/>
+</button>
+
+// This works but might not be the best solution, see below
+<button aria-label="Go to home" class="btn-icon">
+  <Home/>
+</button>
+
+// Do this instead
+<button class="btn-icon">
+  <Home/>
+  <span class="visually-hidden">Go to home</span>
+</button>
+```
+
+:::
+
+## A note on `aria-label`
+
+Although you could provide accessible labels to your elements via the `aria-label` attribute, we
+generally recommend avoiding this and instead suggest that you use your chosen CSS framework's "
+visually hidden" utility whenever possible. You can
+[read more about why `aria-label` might not be the best solution](https://gomakethings.com/revisting-aria-label-versus-a-visually-hidden-class/).
+
```

Comment:
```suggestion

### Example - Radix UI

https://www.radix-ui.com/primitives/docs/utilities/accessible-icon

***tsx
import { ArrowRightIcon } from 'lucide-react';
import { AccessibleIcon } from '@radix-ui/react-accessible-icon';

<AccessibleIcon label="Next item">
  <ArrowRightIcon />
</AccessibleIcon>
***
```

Reply:
https://www.radix-ui.com/primitives/docs/utilities/accessible-icon

Ought to be a link with descriptive text.
````suggestion
### Example - Radix UI

Use [Radix UI's built-in accessible icon utility component](https://www.radix-ui.com/primitives/docs/utilities/accessible-icon).

```tsx
import { ArrowRightIcon } from 'lucide-react';
import { AccessibleIcon } from '@radix-ui/react-accessible-icon';

<AccessibleIcon label="Next item">
  <ArrowRightIcon />
</AccessibleIcon>
```
````